### PR TITLE
feat(dop,pipeline): Project pipeline supports branch search

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Project pipeline supports branch search

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTQ2XSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=292973&iterationID=1146&pId=292482&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     Project pipeline supports branch search         |
| 🇨🇳 中文    |        项目流水线支持分支搜索      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/core/pipeline/definition/definition.proto [hunk 1]
```diff
@@ -42,6 +42,11 @@ service DefinitionService {
       get: "/api/pipeline-definitions/actions/statics-group-by-remote",
     };
   }
+  rpc ListRef (PipelineDefinitionRefListRequest) returns (PipelineDefinitionRefListResponse) {
+    option (google.api.http) = {
+      get: "/api/pipeline-definitions/actions/list-ref",
```
Comment (Effet) [inline api/proto/core/pipeline/definition/definition.proto:47]: I suggest `/api/pipeline-definitions/actions/list-used-refs`

File: modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/condition.go [hunk 2]
```diff
@@ -135,3 +142,27 @@ func (p *CustomFilter) AppCondition() (*model.SelectCondition, error) {
...
+		IdentityInfo: apistructs.IdentityInfo{UserID: p.sdk.Identity.UserID},
+	})
+	if err != nil {
+		return nil, err
+	}
+	cond := model.NewSelectCondition("branch", cputil.I18n(p.sdk.Ctx, "branch"), func() []model.SelectOption {
+		selectOptions := make([]model.SelectOption, 0, len(branches))
+		for _, v := range branches {
+			selectOptions = append(selectOptions, *model.NewSelectOption(func() string {
+				return v
```
Comment (Effet) [inline modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/condition.go:158]: why not just `*model.NewSelectOption(v, ...` ?

File: modules/dop/providers/projectpipeline/deftype/list.go [hunk 3]
```diff
@@ -54,3 +54,20 @@ func (p *ProjectPipelineList) Validate() error {
 
 type ProjectPipelineListResult struct {
 }
+
+type ProjectPipelineRefList struct {
```
Comment (Effet) [inline modules/dop/providers/projectpipeline/deftype/list.go:58]: I suggest add `Ref` -> `UsedRef` is more precise.

File: modules/dop/providers/projectpipeline/service.go [hunk 4]
```diff
@@ -57,6 +57,7 @@ func (p *ProjectPipelineService) WithPermissionSvc(permission *permission.Permis
 type Service interface {
 	Create(ctx context.Context, params *pb.CreateProjectPipelineRequest) (*pb.CreateProjectPipelineResponse, error)
 	List(ctx context.Context, params deftype.ProjectPipelineList) ([]*dpb.PipelineDefinition, int64, error)
+	ListRef(ctx context.Context, params deftype.ProjectPipelineRefList) ([]string, error)
```
Comment (Effet) [inline modules/dop/providers/projectpipeline/service.go:60]: I suggest add Ref -> UsedRef is more precise.

File: modules/pipeline/providers/definition/pipeline_definition.go [hunk 5]
```diff
@@ -273,3 +273,14 @@ func (p pipelineDefinition) StaticsGroupByRemote(ctx context.Context, request *p
 	}
 	return &pb.PipelineDefinitionStaticsResponse{PipelineDefinitionStatistics: pipelineDefinitionStatistics}, nil
 }
+
+func (p pipelineDefinition) ListRef(ctx context.Context, req *pb.PipelineDefinitionRefListRequest) (*pb.PipelineDefinitionRefListResponse, error) {
```
Comment (Effet) [inline modules/pipeline/providers/definition/pipeline_definition.go:277]: I suggest add Ref -> UsedRef is more precise.

**Review by Effet — CHANGES_REQUESTED** [review]

File: api/proto/core/pipeline/definition/definition.proto (hunk 1 above)
Comment (littlejiancc) [reply api/proto/core/pipeline/definition/definition.proto:47]: done

File: modules/dop/providers/projectpipeline/deftype/list.go (hunk 3 above)
Comment (littlejiancc) [reply modules/dop/providers/projectpipeline/deftype/list.go:58]: done

File: modules/dop/providers/projectpipeline/service.go (hunk 4 above)
Comment (littlejiancc) [reply modules/dop/providers/projectpipeline/service.go:60]: done

File: modules/pipeline/providers/definition/pipeline_definition.go (hunk 5 above)
Comment (littlejiancc) [reply modules/pipeline/providers/definition/pipeline_definition.go:277]: done

File: modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/condition.go (hunk 2 above)
Comment (littlejiancc) [reply modules/dop/component-protocol/components/project-pipeline/myPage/tabsTable/filterContainer/customFilter/condition.go:158]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**